src/main/kotlin/io/vertx/gradle/VertxPlugin.kt

```diff
@@ -177,5 +179,48 @@ class VertxPlugin : Plugin<Project> {
 
     logger.debug("The vertxRun task has been configured")
   }
+
+  private fun configureVertxDebugTask(project: Project) {
+    val vertxExtension = project.vertxExtension()
+    val javaConvention = project.convention.getPlugin(JavaPluginConvention::class.java)
+    val mainSourceSet = javaConvention.sourceSets.getByName("main")
+
+    (project.tasks.getByName("vertxDebug") as JavaExec).apply {
+      group = "Application"
+      description = "Debugs this project as a Vert.x application"
+
+      workingDir(File(vertxExtension.workDirectory))
+      jvmArgs(vertxExtension.jvmArgs)
+      jvmArgs(computeDebugOptions(project))
+      classpath(mainSourceSet.runtimeClasspath)
+
+      main = if (vertxExtension.redeploy) "io.vertx.core.Launcher" else vertxExtension.launcher
+
+      if (vertxExtension.launcher == "io.vertx.core.Launcher") {
+        if (vertxExtension.mainVerticle.isBlank()) {
+          throw GradleException("Extension property vertx.mainVerticle must be specified when using io.vertx.core.Launcher as a launcher")
+        }
+        args("run", vertxExtension.mainVerticle)
+      }
+
+      if (vertxExtension.config.isNotBlank()) {
+        args("--conf", vertxExtension.config)
+      }
+      vertxExtension.args.forEach { args(it) }
+    }
+
+    logger.debug("The vertxRun task has been configured")
+  }
+
+  private fun computeDebugOptions(project: Project): List<String> {
+    val vertxExtension = project.vertxExtension()
+    val debugger = "-agentlib:jdwp=transport=dt_socket,server=y,suspend=" +
+      (if (vertxExtension.debugSuspend) "y" else "n") + ",address=" + vertxExtension.debugPort
+    val disableEventLoopchecker = "-Dvertx.options.maxEventLoopExecuteTime=" + java.lang.Long.MAX_VALUE
```

Comment:
Good catch.

This should be documented BTW, like adding a sentence in the README note you added.